flink-end-to-end-tests/flink-end-to-end-tests-elasticsearch6/src/test/java/com/apache/flink/streaming/tests/Elasticsearch6SinkExternalContext.java

```diff
@@ -0,0 +1,135 @@
+
+import java.io.IOException;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Random;
+import java.util.concurrent.ThreadLocalRandom;
+
+class Elasticsearch6SinkExternalContext
+        implements DataStreamSinkV2ExternalContext<TupleC2<String, String>> {
+
+    private static final Logger LOG =
+            LoggerFactory.getLogger(Elasticsearch6SinkExternalContext.class);
+
+    private static final String INDEX_NAME_PREFIX = "es-index";
+    private static final int RANDOM_STRING_MAX_LENGTH = 50;
+    private static final int NUM_RECORDS_UPPER_BOUND = 500;
+    private static final int NUM_RECORDS_LOWER_BOUND = 100;
+    private static final int PAGE_LENGTH = 100;
+
+    protected final String indexName;
+
+    private final String addressInternal;
+    private final List<URL> connectorJarPaths;
+    private final RestHighLevelClient client;
+
+    Elasticsearch6SinkExternalContext(
+            String addressExternal, String addressInternal, List<URL> connectorJarPaths) {
+        this.addressInternal = addressInternal;
+        this.connectorJarPaths = connectorJarPaths;
+        this.indexName =
+                INDEX_NAME_PREFIX + "-" + ThreadLocalRandom.current().nextLong(Long.MAX_VALUE);
+        HttpHost httpHost = HttpHost.create(addressExternal);
+        RestClientBuilder restClientBuilder = RestClient.builder(httpHost);
+        this.client = new RestHighLevelClient(restClientBuilder);
+    }
+
+    @Override
+    public Sink<TupleC2<String, String>> createSink(TestingSinkSettings sinkSettings) {
+        return new Elasticsearch6SinkBuilder<TupleC2<String, String>>()
+                .setHosts(HttpHost.create(this.addressInternal))
+                .setEmitter(new Elasticsearch6TestEmitter(indexName))
+                .setBulkFlushMaxActions(100) // emit after every element, don't buffer
+                .build();
+    }
+
+    @Override
+    public ExternalSystemDataReader<TupleC2<String, String>> createSinkDataReader(
+            TestingSinkSettings sinkSettings) {
+        Elasticsearch6Utils.refreshIndex(client, indexName);
+        return new Elasticsearch6DataReader(client, indexName, PAGE_LENGTH);
+    }
+
+    @Override
+    public List<TupleC2<String, String>> generateTestData(
+            TestingSinkSettings sinkSettings, long seed) {
+        Random random = new Random(seed);
+        List<TupleC2<String, String>> randomStringRecords = new ArrayList<>();
+        int recordNum =
+                random.nextInt(NUM_RECORDS_UPPER_BOUND - NUM_RECORDS_LOWER_BOUND)
+                        + NUM_RECORDS_LOWER_BOUND;
+
+        for (int i = 0; i < recordNum; i++) {
+            int valueLength = random.nextInt(RANDOM_STRING_MAX_LENGTH) + 1;
+            String key = Integer.toString(i);
+            String value = RandomStringUtils.random(valueLength, true, true);
+            randomStringRecords.add(TupleC2.of(key, value));
+        }
+        return randomStringRecords;
```

Comment:
Nit: Again, please use the stream syntax.

Reply:
Copy-paste from Kafka tests. Migrated.